### ISSUE TITLE
Creating a model sometimes returns a numeric value as `.id` 

### ISSUE THREAD

**Issue by mik3y** [mik3y/django-spicy-id#6]
In some environments, creating a model will return an instance where `.id` is a _number_ - even after the fix attempted in c3303d / v0.6.0. 

This is a bug; consumers of spicy ids should always expect & get a string back, never the underlying numeric value.

Example from v0.6.1:
```
>>> o = SomeModelWhereRandomizeIsEnabled()
>>> o.save()
>>> o.id
3735928559
>>> o.refresh_from_db()
>>> o.id
'ex_deadbeef`
```

**Comment by mik3y** [comment]
I believe the culprit is this block in Django's `db.models.sql.compiler`:

https://github.com/django/django/blob/fbd16438f46bc2128926958ad24331da5d1b406f/django/db/models/sql/compiler.py#L1825-L1847

Specially, when neither `self.connection.features.can_return_rows_from_bulk_insert` nor `self.connection.features.can_return_columns_from_insert` are returned, the result of `execute_sql` is the value of `self.connection.ops.last_insert_id(..)`

This block is called during `_save_table`, which takes its result and does as `setattr` on the instance:

https://github.com/django/django/blob/fbd16438f46bc2128926958ad24331da5d1b406f/django/db/models/base.py#L1057-L1063

I _think_ it would be more correct for Django to flow the output of `last_insert_id()` through the custom field's `from_db_value()`. But there probably aren't a lot of other folks creating wacky "virtual" primary keys like we are necessitating any transformation here.

**Comment by mik3y** [comment]
Ticketed upstream as: https://code.djangoproject.com/ticket/34792

**Comment by mik3y** [comment]
Ya gotta love Django: The issue was previously identified & fixed in Django 4.x: https://code.djangoproject.com/ticket/32442

I will suggest (to myself!) that we add a usage warning for django < 4 & check in the workaround from #8 .

**Comment by mik3y** [comment]
Fixed in #8 / v0.7.0.
